### PR TITLE
GraphQL API and UI for ownership analytics

### PR DESCRIPTION
**Pull request by cbart**
Part of https://github.com/sourcegraph/sourcegraph/issues/52826

Admin analytics page for own with a zero-value graphQL API.

Notes:

- Use `own-analytics` feature flag to enable the panel in the Admin UI.

**Screenshots**

Bare bones UI:
Screenshot 2023-06-03 at 9 21 45 PM

When analytics job is not enabled (default):
Screenshot 2023-06-03 at 9 02 58 PM

## Test plan

Visual assessment.

### CODE REVIEW

**Comment by cbart** [comment]
This is a draft, but please take a look. We're going to ship this analytics panel without data first, and then include backend works.

File: client/web/src/enterprise/own/admin-ui/OwnAnalyticsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    ownershipReasonType: string
+    entriesCount: number
+    fill: string
+}
+
+export const OwnAnalyticsPage: FC = () => {
+    // TODO: Error handling and loading
+    const { data } = useQuery<GetOwnSignalConfigurationsResult>(GET_OWN_JOB_CONFIGURATIONS, {})
+    const enabled =
+        data?.ownSignalConfigurations.some(
```
Comment (sashaostrikov) [inline client/web/src/enterprise/own/admin-ui/OwnAnalyticsPage.tsx:26]: maybe `find` would be better here? We actually need to find the only `analytics` config.

File: client/web/src/enterprise/own/admin-ui/OwnAnalyticsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            </Card>
+            <Text className="font-italic text-center mt-2">
+                All events are generated from entries in the event logs table and are updated every 24 hours.
+            </Text>
+        </>
+    )
+}
+
+const OwnEnableAnalytics: FC = () => {
+    return <Text>Analytics is not enabled, please enable Own analytics job first to see own stats.</Text>
```
Comment (sashaostrikov) [inline client/web/src/enterprise/own/admin-ui/OwnAnalyticsPage.tsx:85]: maybe? to be in style with `Own analytics`
```suggestion
    return <Text>Analytics is not enabled, please enable Own analytics job first to see Own stats.</Text>
```

File: client/web/src/enterprise/own/admin-ui/OwnAnalyticsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    return enabled ? <OwnAnalyticsPanel /> : <OwnEnableAnalytics />
+}
+
+const OwnAnalyticsPanel: FC = () => {
+    // TODO: Error handling and loading
+    const { data } = useQuery<GetInstanceOwnStatsResult>(GET_INSTANCE_OWN_STATS, {})
+
+    const ownSignalsData: OwnUsageDatum[] = [
+        {
+            ownershipReasonType: 'Codeowned files',
```
Comment (sashaostrikov) [inline client/web/src/enterprise/own/admin-ui/OwnAnalyticsPage.tsx:38]: `codeowned` sounds funny, but I cannot come up with a better option myself 🤷🏻

File: cmd/frontend/graphqlbackend/own.graphql [hunk 4]
```diff
@@ -120,6 +137,7 @@ type Ownership {
 The only way we can recognize ownership at this point is through CODEOWNERS file entry.
 """
 enum OwnershipReasonType {
+    ASSIGNED_OWNER
```
Comment (sashaostrikov) [inline cmd/frontend/graphqlbackend/own.graphql:140]: lol, it was forgotten? 😁 (that's me who forgot to add it)

File: enterprise/cmd/frontend/internal/own/resolvers/resolvers.go [hunk 5]
```diff
@@ -223,6 +224,18 @@ func (r *ownResolver) GitTreeOwnership(
 	return r.ownershipConnection(args, ownerships)
 }
 
+func (r *ownResolver) GitTreeOwnershipStats(ctx context.Context, tree *graphqlbackend.GitTreeEntryResolver) (graphqlbackend.OwnershipStatsResolver, error) {
+	return &ownStatsResolver{
```
Comment (sashaostrikov) [inline enterprise/cmd/frontend/internal/own/resolvers/resolvers.go:228]: I was about to write about nil pointer possibility, but then I saw that we don't check for it anywhere in this file. So I guess it's kinda guaranteed.

**Review by sashaostrikov — APPROVED** [review]
LET'S GO!

File: client/web/src/enterprise/own/admin-ui/OwnAnalyticsPage.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    GetInstanceOwnStatsResult,
+    GetOwnSignalConfigurationsResult,
+    OwnSignalConfig,
+} from '../../../graphql-operations'
+import { AnalyticsPageTitle } from '../../../site-admin/analytics/components/AnalyticsPageTitle'
+import { ChartContainer } from '../../../site-admin/analytics/components/ChartContainer'
+
+import { GET_INSTANCE_OWN_STATS, GET_OWN_JOB_CONFIGURATIONS } from './query'
+
+interface OwnCovaerateDatum {
```
Comment (sashaostrikov) [inline client/web/src/enterprise/own/admin-ui/OwnAnalyticsPage.tsx:16]: should it be like this, or I didn't get it? :)
```suggestion
interface OwnCovariateDatum {
```

File: cmd/frontend/graphqlbackend/git_tree_entry.go [hunk 7]
```diff
@@ -389,6 +389,17 @@ func (r *GitTreeEntryResolver) Ownership(ctx context.Context, args ListOwnership
 	return nil, nil
 }
 
+type OwnershipStatsArgs struct {
+	Reasons *[]OwnershipReasonType
+}
+
+func (r *GitTreeEntryResolver) OwnershipStats(ctx context.Context) (OwnershipStatsResolver, error) {
+	if _, ok := r.ToGitTree(); !ok {
+		return nil, nil // TODO: No support for blob for now, but can add it.
```
[the remark this answers is not in the document]
Comment (cbart) [reply cmd/frontend/graphqlbackend/git_tree_entry.go:398]: Nope, we don't need blob support tracked. I'll change the comment.

File: client/web/src/enterprise/own/admin-ui/OwnAnalyticsPage.tsx (hunk 6 above)
Comment (cbart) [reply client/web/src/enterprise/own/admin-ui/OwnAnalyticsPage.tsx:16]: Ooops. Good catch. I was meant to be coverage.

**Review by cbart — COMMENTED** [review]
Thank you!

File: client/web/src/enterprise/own/admin-ui/OwnAnalyticsPage.tsx (hunk 1 above)
Comment (sashaostrikov) [reply client/web/src/enterprise/own/admin-ui/OwnAnalyticsPage.tsx:27]: that's nice!

**Pull request merged**